src/rebase.js

```diff
@@ -116,13 +116,22 @@ module.exports = (function(){
   function _fetch(endpoint, options){
     _validateEndpoint(endpoint);
     optionValidators.context(options);
-    optionValidators.then(options);
     options.queries && optionValidators.query(options);
     var ref = firebase.database().ref(endpoint);
     ref = _addQueries(ref, options.queries);
-    ref.once('value', (snapshot) => {
+    return ref.once('value').then(snapshot => {
       var data = options.asArray === true ? _toArray(snapshot) : snapshot.val();
-      options.then.call(options.context, data);
+      if(options.then){
+        options.then.call(options.context, data);
+      }
+      return data;
+    }, err => {
+      //call onFailure callback if it exists otherwise return a rejected promise
+      if(options.onFailure && typeof options.onFailure === 'function'){
+        options.onFailure.call(options.context, err);
+      } else {
```

Comment:
Removing the `else` will result in this error when not using a `.catch`  and I think we need to avoid that to not break people's existing apps. 

![screen shot 2016-09-14 at 8 54 20 am](https://cloud.githubusercontent.com/assets/2364926/18519578/4fc644d0-7a59-11e6-9ed7-bd7d4877287b.png)

You said before that it is kind of weird that both exist in the firebase SDK but pointed out that it is technically valid to use at the same time. I agree with that, but I can't really think of a use case where someone would want to use both simultaneously which I think is the only real problem

In my opinion this implementation does provide
> expected behaviour for those who prefer callbacks, and for those who use Promises as well.

This works for the Promise users: 
```javascript
fetch().then().catch()
```

And this works for the callback users (without the thrown exception):

```javascript
fetch('endpoint', {
  then: onSuccess,
  onFailure: onFailure
})
```

Would you agree with that? I still think making it align as close to the firebase api as possible is worth it, and am definitely not opposed to changing this so it does that.